neural_compressor/compression/pruner/criteria.py

```diff
@@ -96,7 +99,9 @@ def on_step_begin(self):
         """Calculate and store the pruning scores based on a magnitude criterion."""
         with torch.no_grad():
             for key in self.modules.keys():
-                p = self.modules[key].weight.data
+                param = self.modules[key].weight
+                data = safe_get_data(param)
+                p = data
```

Comment:
p =  safe_get_data(param)

Reply:
Thanks, updated in [ed6395b](https://github.com/intel/neural-compressor/pull/1492/commits/ed6395b2fa9ae9aecce68b2a0ceb6e3e1ec2f02d).